### DISCUSSION TITLE
how do I pass "resource" in  RequestBody while creating OAuthFlows() using clientCredentials?

### DISCUSSION THREAD

**Discussion by rajakumare1** [swagger-api/swagger-ui#8933] [Q&A]
Hi All, 

I understand  how to setup a OAUTH2 ClientCredentials while creating  new OpenAPI().
 I  am trying to authenticate with Azure AD roles(not scopes) with client crendentials. I need to pass **resource** as one of the parameter in the Body. How do I do that? Please advice

new OAuthFlows()
                        .clientCredentials(new OAuthFlow()
                        .tokenUrl(<url>).scopes(scopes()
                        )

![Screenshot 2023-06-15 at 9 52 31 AM](https://github.com/swagger-api/swagger-ui/assets/19310375/c3c154a1-7777-4c98-87ae-9656a76984e1)

**Comment by cihatsolak** [comment]
Hi,

In addition to the clientCredentials() method in OAuthFlows, you can add the "resource" parameter using the additionalQueryStringParams() method.

```
OAuthFlows flows = new OAuthFlows()
    .clientCredentials(new OAuthFlow()
        .tokenUrl("https://blablabla/token")
        .scopes(scopes())
        .additionalQueryStringParams(Collections.singletonMap("resource", "resource-value")));
```
This way you can create OAuthFlows and OAuthFlow objects and use the clientCredentials flow when requesting the "resource" parameter.
